src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -866,10 +865,15 @@ internal sealed class LimitMemoryStream : MemoryStream
         {
             private readonly int _maxSize;
 
-            public LimitMemoryStream(int maxSize, int capacity)
-                : base(capacity)
+            public LimitMemoryStream(int maxSize, long capacity)
+                : base(0)
```

Comment:
I have verified, by reading code, that `(new MemoryStream(0)).Capacity = c` vs `new MemoryStream(c)` have negligent affect on performance.